ako-infra/ingestion/avi_infra_controller.go

```diff
@@ -73,6 +77,137 @@ func (a *AviControllerInfra) VerifyAviControllerLicense() error {
 	return nil
 }
 
+func (a *AviControllerInfra) DeriveCloudNameAndSEGroupTmpl(tz string) (error, string, string) {
+	// This method queries the Avi controller for all available cloud and then returns the cloud that matches the supplied transport zone
+	uri := "/api/cloud/"
+	result, err := lib.AviGetCollectionRaw(a.AviRestClients.AviClient[0], uri)
+	if err != nil {
+		utils.AviLog.Errorf("Get uri %v returned err %v", uri, err)
+		return err, "", ""
+	}
+
+	elems := make([]json.RawMessage, result.Count)
+	err = json.Unmarshal(result.Results, &elems)
+	if err != nil {
+		utils.AviLog.Errorf("Failed to unmarshal data, err: %v", err)
+		return err, "", ""
+	}
+	for i := 0; i < len(elems); i++ {
+		cloud := models.Cloud{}
+		err = json.Unmarshal(elems[i], &cloud)
+		if err != nil {
+			utils.AviLog.Warnf("Failed to unmarshal cloud data, err: %v", err)
+			continue
+		}
+		vtype := *cloud.Vtype
+		if vtype == lib.CLOUD_NSXT && cloud.NsxtConfiguration != nil {
+			if cloud.NsxtConfiguration.TransportZone != nil && *cloud.NsxtConfiguration.TransportZone == tz {
+				utils.AviLog.Infof("Found NSX-T cloud :%s match Transport Zone : %s", *cloud.Name, tz)
+				if *cloud.SeGroupTemplateRef != "" {
+					tokenized := strings.Split(*cloud.SeGroupTemplateRef, "/api/serviceenginegroup/")
+					if len(tokenized) == 2 {
+						return nil, *cloud.Name, tokenized[1]
+					}
+				}
+			}
+			return nil, *cloud.Name, ""
+		}
+	}
+	return errors.New("Cloud not found"), "", ""
+}
+
+func (a *AviControllerInfra) SetupSEGroup(tz string) bool {
+	// This method checks if the cloud in Avi has a SE Group template configured or not. If has the SEG template then it returns true, else false
+	err, _, segUuid := a.DeriveCloudNameAndSEGroupTmpl(tz)
+	if err == nil {
+		if segUuid == "" {
```

Comment:
I think we can optimise this code as (only change uri on segUuid .. rest of code of fetching seg and configuring it is common):
if segUuid == "" {
  uri = "/api/serviceenginegroup/?name=default-group"
} else {
  uri = "/api/serviceenginegroup/" + segUuid"
}
result, err := lib.AviGetCollectionRaw(a.AviRestClients.AviClient[0], uri)
.
.